fundingmanager_test.go

```diff
@@ -3019,6 +3020,91 @@ func TestFundingManagerFundAll(t *testing.T) {
 	}
 }
 
+// TestFundingManagerFundMax tests that we can initiate a funding request to
+// use the maximum allowed funds remaining in the wallet.
+func TestFundingManagerFundMax(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		coins  []*lnwallet.Utxo
+		change bool
+	}{
+		{
+			// We will spend all the funds in the wallet, and
+			// expects no change output.
+			coins: []*lnwallet.Utxo{{
+				AddressType: lnwallet.WitnessPubKey,
+				Value:       MaxBtcFundingAmount + 1,
+				PkScript:    coinPkScript,
+				OutPoint: wire.OutPoint{
+					Hash:  chainhash.Hash{},
+					Index: 0,
+				},
+			}},
+			change: false,
+		},
+		{
+			// We spend less than the funds in the wallet,
+			// so a change output should be created.
+			coins: []*lnwallet.Utxo{{
+				AddressType: lnwallet.WitnessPubKey,
+				Value:       2 * MaxBtcFundingAmount,
+				PkScript:    coinPkScript,
+				OutPoint: wire.OutPoint{
+					Hash:  chainhash.Hash{},
+					Index: 0,
+				},
+			}},
+			change: true,
+		},
+	}
+
+	for _, test := range tests {
+		alice, bob := setupFundingManagers(t)
+		defer tearDownFundingManagers(t, alice, bob)
+
+		// We set up our mock wallet to control a list of UTXOs that sum to
+		// more than the max channel size.
+		alice.fundingMgr.cfg.Wallet.WalletController.(*mockWalletController).utxos = test.coins
```

Comment:
The new commits does that now. The only difference is that inside `setupFundingManagers()` both `alice` and `bob` will then get funds, whereas this would only apply to `alice`. In the existing `TestFundingManagerFundAll` inside this file the same technique is used. If you agree I would also change it there?

Reply:
Ah, I see. Didn't look at the other tests in this file at all :sweat_smile: It should be fine to give both nodes some UTXOs, as long as it isn't a dual-funding test only Alice should use them anyway.
Feel free to use the following form (shouldn't use additional memory, the compiler optimizes things like this):
```
		addFunds := func(fundingCfg *fundingConfig) {
			wc := fundingCfg.Wallet.WalletController
			wc.(*mockWalletController).utxos = test.coins
		}
```